substrate/frame/assets-freezer/src/types.rs

```diff
@@ -0,0 +1,36 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use frame_support::{
+	pallet_prelude::{Decode, Encode, MaxEncodedLen, RuntimeDebug, TypeInfo},
+	traits::fungibles::Inspect,
+};
+
+pub type AssetIdOf<T, I> = <pallet_assets::Pallet<T, I> as Inspect<AccountIdOf<T>>>::AssetId;
+pub type AssetBalanceOf<T, I> = <pallet_assets::Pallet<T, I> as Inspect<AccountIdOf<T>>>::Balance;
+pub type AccountIdOf<T> = <T as frame_system::Config>::AccountId;
```

Comment:
Better make it publicly accessible from `pallet-assets`. I tried attaching to the types `AssetId` and `Balance` from `fungibles::Inspect`. However, found out that it was not enough without having an structure (`T` couldn't be) which implemented that trait in first place.

Now that I see it, another possibility (since we already implemented that here) is to just refer as `Pallet<T, I>` instead of looking for `pallet_assets::Pallet<T, I>`. Yet, a re-export would be better in terms of getting (just the right amount) of less coupling between those.

Reply:
This pallet is already tightly coupled with pallet-assets so that's ok with me